server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -0,0 +1,111 @@
+import org.apache.james.user.api.UsersRepository
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Mono
+import reactor.core.scala.publisher.SFlux
+import reactor.netty.http.server.HttpServerRequest
+
+import scala.compat.java8.StreamConverters._
+
+object UserCredential {
+  type BasicAuthenticationHeaderValue = String Refined MatchesRegex["Basic [\\d\\w=]++"]
+  type CredentialsAsString = String Refined MatchesRegex[".*:.*"]
+
+  private val logger = LoggerFactory.getLogger(classOf[UserCredential])
+  private val BASIC_AUTHENTICATION_PREFIX: String = "Basic "
+
+  def parseUserCredentials(token: String): Option[UserCredential] = {
+    val refinedValue: Either[String, BasicAuthenticationHeaderValue] = refineV(token)
+
+    refinedValue match {
+      // Ignore Authentication headers not being Basic Auth
+      case Left(_) => None
+      case Right(value) => extractUserCredentialsAsString(value)
+    }
+  }
+
+  private def extractUserCredentialsAsString(token: BasicAuthenticationHeaderValue): Option[UserCredential] = {
+    val encodedCredentials = token.replace(BASIC_AUTHENTICATION_PREFIX, "")
+    val decodedCredentialsString = new String(Base64.getDecoder.decode(encodedCredentials))
+    val refinedValue: Either[String, CredentialsAsString] = refineV(decodedCredentialsString)
+
+    refinedValue match {
+      case Left(errorMessage: String) =>
+        logger.info(s"Supplied basic authentication credentials do not match expected format. $errorMessage")
+        None
+      case Right(value) => toCredential(value)
+    }
+  }
+
+  private def toCredential(token: CredentialsAsString): Option[UserCredential] = {
+    val partSeparatorIndex: Int = token.indexOf(':')
+    val usernameString: String = token.substring(0, partSeparatorIndex)
+    val passwordString: String = token.substring(partSeparatorIndex + 1)
+
+    try {
+      Some(UserCredential(Username.of(usernameString), passwordString))
+    } catch {
+      case throwable: IllegalArgumentException => {
+        logger.info("Username is not valid", throwable)
+        None
+      }
+      case unexpectedException => {
+        logger.error("Unexpected Exception", unexpectedException)
+        None
+      }
+    }
+  }
+}
+
+case class UserCredential(username: Username, password: String)
+
+class BasicAuthenticationStrategy @Inject()(val usersRepository: UsersRepository,
+                                            val mailboxManager: MailboxManager) extends AuthenticationStrategy {
+
+  override def createMailboxSession(httpRequest: HttpServerRequest): Mono[MailboxSession] = {
+    SFlux.fromStream(() => authHeaders(httpRequest).toScala[Stream])
+      .map(parseUserCredentials)
+      .handle(publishNext)
+      .filter(isValid)
+      .map(_.username)
+      .map(mailboxManager.createSystemSession)
+      .singleOrEmpty()
+      .asJava()
+  }
+
+  private def publishNext[T]: (Option[T], reactor.core.publisher.SynchronousSink[T]) => scala.Unit =
```

Comment:
why do you need to prefix `Unit` with `scala` ?

Reply:
I bet it mean: return a generic value for a generic method i created.

Reply:
my question was just why do you prefix Unit with the `scala` package ? it should not be needed.

Reply:
I pushed a fixup